File: samueltm/boredom/math/geometry/Cylinder.java
package samueltm.boredom.math.geometry;

public class Cylinder extends Shape {
    private final double height;
    private final double radius;

    public Cylinder(double height, double diameter) {
        this.height = height;
        this.radius = diameter / 2;
    }

    @Override
    public double getArea() {
        return (2 * Math.PI * Math.pow(radius, 2)) + (2 * Math.PI * radius * height);
    }
}

File: samueltm/boredom/math/geometry/Cylinder.kt
package samueltm.boredom.math.geometry

import kotlin.math.PI
import kotlin.math.pow

class Cylinder(private val height: Double, diameter: Double) : Shape() {
    private val radius: Double = diameter / 2

    override fun getArea() = (2 * PI * radius.pow(2)) + (2 * PI * radius * height)
}